lib/browser/crypto.ts

```diff
@@ -13,4 +13,48 @@
  * permissions and limitations under the License.
  */
 
-export const dummy : boolean = false;
+import * as Crypto from "crypto-js";
+import { Hashable } from "../common/crypto";
+
+export class Md5Hash {
+    private hash?: Crypto.WordArray;
+    
+    update(data: Hashable) {
+        this.hash = Crypto.MD5(data.toString(), this.hash ? this.hash.toString() : undefined);
+    }
+
+    digest(truncate_to?: number): DataView {
+        const digest = this.hash ? this.hash.toString(Crypto.enc.Utf8) : '';
+        const truncated = digest.substring(0, truncate_to ? truncate_to : digest.length);
+        const encoder = new TextEncoder();
+        const bytes = encoder.encode(truncated);
+        return new DataView(bytes);
+    }
+}
+
+export function hash_md5(data: Hashable, truncate_to?: number): DataView {
+    const md5 = new Md5Hash();
+    md5.update(data);
+    return md5.digest();
```

Comment:
Does truncate_to get magically used here?

Reply:
No, it gets magically forgotten by the implementer, thanks!